qt: Add option to uninstall a game.

Steveice10 opened the pull request:
Adds an option to the Qt game list to uninstall any installed title. Closes https://github.com/citra-emu/citra/issues/5386

Menu item conditions:
* Everything is enabled if the title is installed, has an installed update, or has an installed DLC.
* Game is enabled if the title is installed.
* Update is enabled if the title has an installed update.
* DLC is enabled if the title has an installed DLC.

New UI elements:

Menus:
<img width="322" alt="Screenshot 2023-10-12 at 2 41 24 AM" src="https://github.com/citra-emu/citra/assets/1269164/c5539275-8506-4807-a5fc-17783297bd69">

Confirmations:
<img width="272" alt="Screenshot 2023-10-12 at 2 41 31 AM" src="https://github.com/citra-emu/citra/assets/1269164/72f45c49-32b1-47e0-affd-f6b601289582">
<img width="275" alt="Screenshot 2023-10-12 at 2 41 37 AM" src="https://github.com/citra-emu/citra/assets/1269164/15418140-ccec-44d1-80be-2cd89490b5e4">
<img width="273" alt="Screenshot 2023-10-12 at 2 41 47 AM" src="https://github.com/citra-emu/citra/assets/1269164/d3c3f5e2-4e28-43cf-b5f3-aaf5fb5a2a96">
<img width="269" alt="Screenshot 2023-10-12 at 2 41 56 AM" src="https://github.com/citra-emu/citra/assets/1269164/0452cd43-0f4f-4363-bf76-6ddcf5ffe1c8">

Success:
<img width="267" alt="Screenshot 2023-10-12 at 6 41 31 PM" src="https://github.com/citra-emu/citra/assets/1269164/535872af-3e62-4195-b5a9-8a415051511b">

Failure:
<img width="269" alt="Screenshot 2023-10-12 at 6 43 32 PM" src="https://github.com/citra-emu/citra/assets/1269164/05cd645b-4552-4611-8c86-f4d527f503aa">

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/citra-emu/citra/7064)
<!-- Reviewable:end -->

FearlessTobi commented on src/citra_qt/game_list.cpp:
```diff
@@ -641,8 +668,89 @@ void GameList::AddGamePopup(QMenu& context_menu, const QString& path, u64 progra
+            const auto media_type = std::get<Service::FS::MediaType>(title);
+            const auto program_id = std::get<u64>(title);
+
+            emit main_window->UninstallReport(UninstallStatus::Started, name);
```
maybe the whole function for removing titles should just be in the main window instead?
then we wouldnt need this helper function

FearlessTobi commented on src/citra_qt/main.cpp:
```diff
@@ -1750,6 +1751,21 @@ void GMainWindow::OnCIAInstallFinished() {
+        this->statusBar()->showMessage(tr("Uninstalling %1").arg(name));
+        break;
+    case UninstallStatus::Succeeded:
+        this->statusBar()->showMessage(tr("%1 has been uninstalled successfully.").arg(name));
```
maybe this should be a popup too?
i don't know if our users will see it otherwise :p

FearlessTobi reviewed (approved):
Thanks for implementing this!

Steveice10 commented:
Addressed comments and reworked the flow a bit. The signals are no longer needed, we use QtConcurrent::map, and it has a progress bar for how many titles have been uninstalled (up to three total possible - game, update, and DLC).
